soroban-env-common/env.json

```diff
@@ -2042,7 +2042,16 @@
                     "args": [],
                     "return": "Val",
                     "docs": "A dummy function taking 0 arguments and performs no-op. This function is for test purpose only, for measuring the roundtrip cost of invoking a host function, i.e. host->Vm->host."
-                }
+                },
+                {
+                    "export": "0",
+                    "name": "protocol_gated_dummy",
+                    "args": [],
+                    "return": "Val",
+                    "docs": "A dummy function for testing the protocol gating. Takes 0 arguments and performs no-op",
+                    "min_supported_protocol": 19,
```

Comment:
Isn't this a protocol change? I thought it was only intended for testing. I think it should still be gated by protocol 21, or otherwise we would risk a divergence.

Reply:
This won't have an effect on protocol because no contract can link to it. It's min and max protocol are both specified as 19. So if a contract tries, it will fail the validation check during the upload (same as linking to a non-existent function today).

The reason this has to be 19 (or any value lower than 20) is because the test includes the case where the ledger protocol version > host function max protocol (see `test_host_protocol_gating_for_wasm`), and the ledger protocol version cannot be larger than the env version (20, which is compiled in and non-overridable).

Note setting the min_supported_protocol to 21 is using the exact same mechanism for protection against contract built with protocol 20. So effectively this function is always protocol gated out.

Reply:
Ah, that's clever. Thanks for the explanation.

Reply:
Good point. I've added a comment to the function docs in https://github.com/stellar/rs-soroban-env/pull/1356